tardis/conftest.py

```diff
@@ -32,8 +50,48 @@ def pytest_addoption(parser):
     parser.addoption("--atomic-dataset", dest='atomic-dataset', default=None,
                      help="filename for atomic dataset")
 
-    parser.addoption("--slow-test-data", dest="slow-test-data",
-                     help="path to directory having baseline data for slow tests")
+    parser.addoption("--integration-tests", dest="integration-tests",
+                     help="path to configuration file for integration tests")
+
+
+def pytest_configure(config):
+    html_file = tempfile.NamedTemporaryFile(delete=False)
+    # Html test report will be generated at this filepath by pytest-html plugin
+    config.option.htmlpath = html_file.name
+
+
+def pytest_unconfigure(config):
+    # Html report created by pytest-html plugin is read here, uploaded to
+    # dokuwiki and finally deleted.
+    if dokuwiki_available:
+        try:
+            doku_conn = dokuwiki.DokuWiki(
+                url=integration_tests_config["dokuwiki"]["url"],
+                user=integration_tests_config["dokuwiki"]["username"],
+                password=integration_tests_config["dokuwiki"]["password"])
+
+            githash = tardis.__githash__
+            report_content = open(config.option.htmlpath, 'rb').read()
+            report_content = report_content.replace("<!DOCTYPE html>", "")
+
+            report_content = (
+                "Test executed on commit "
+                "[[https://www.github.com/tardis-sn/tardis/commit/{0}|{0}]]\n\n"
+                "{1}".format(githash, report_content)
+            )
+            # Upload report on dokuwiki. Temporary link due to prototyping purposes.
+            doku_conn.pages.append("reports:{0}".format(githash[:7]), report_content)
+        except gaierror, dokuwiki.DokuWikiError:
+            pass
```

Comment:
again - stick the things that generate the error in the try except block and nothing else.